### DISCUSSION TITLE
Need a way to revoke an offline token without revoking user client grant or revoking active user session

### DISCUSSION THREAD

**Discussion by harshit-kumar-v2** [keycloak/keycloak#31034] [Q&A]
Requirement
I need to revoke an offline token generated by using scope=offline_access but not revoking any other offline tokens for that user. I also do not want to end active user session corresponding to that offline token.

keycloak behaviour
1. using /revoke endpoint :- When using revoke endpoint (i send the offline token as body param) but this revoke the complete user grant for that user (i.e. all token for that user old and newly generated gets revoked). Basically this ends all sessions and tokens. This is mentioned in doc as
```
Token Revocation endpoint
/realms/{realm-name}/protocol/openid-connect/revoke
The token revocation endpoint is used to revoke tokens. Both refresh tokens and access tokens are supported by this endpoint. When revoking a refresh token, the user consent for the corresponding client is also revoked.

For more details on how to invoke on this endpoint, see [OAuth 2.0 Token Revocation specification](https://datatracker.ietf.org/doc/html/rfc7009).
```

2. using /logout endpoint: There is a way to end a single user session but send a logout request with client id and refresh token as endpoint. This does prevent the other offline tokens from getting revoked but still now full filling my use case. This is what should be expected (but i want to revoke only token not the session).

A bit on how auth0 behaves here
The auth0 revoke endpoints take in the refresh token / offline token and revokes only the token(i.e. we can't refresh using that token any more) this does not do anything to session or other client grant entities. This should be the ideal behaviour of revoke endpoint.

Is there any way to achieve similar result, need help with the use case.

**Comment by MaltheGram** [comment]
@harshit-kumar-v2 Did you ever find a solution for this? I need the exact same

**Reply by nonvoy** [reply]
Me too
